common/app/com/commercetools/sunrise/framework/viewmodels/forms/QueryStringUtils.java

```diff
@@ -50,6 +48,20 @@ private QueryStringUtils() {
     }
 
     /**
+     * Extracts the query string from the request.
+     * @param httpRequest current HTTP request
+     * @param ignoredParams params that should not be included in the link
+     * @return query string from the request
+     */
+    public static Map<String, List<String>> extractQueryString(final Http.Request httpRequest,final Set<String> ignoredParams) {
+
+        Set<String> ignored = Optional.ofNullable(ignoredParams).orElseGet(HashSet::new);
+        return httpRequest.queryString().entrySet().stream()
+                .filter(stringEntry -> !ignored.contains(stringEntry.getKey()))
+                .collect(toMap(Map.Entry::getKey, entry -> asList(entry.getValue())));
```

Comment:
To avoid duplication, I would let the other `extractQueryString` method to call this one with an empty set as `ignoredParams` (i.e. `Collections.emptySet()`). But I'm not sure how efficient would be the stream with a filter predicate that it's always false. Otherwise what we can surely avoid duplicating is the Collector `toMap(Map.Entry::getKey, entry -> asList(entry.getValue()))`. That would be enough to avoid duplication, I leave it to your decision :)